
project system/core/
diff --git a/system/core/init/init.c b/system/core/init/init.c
index 1cb9194..5ce3b96 100644
--- a/system/core/init/init.c
+++ b/system/core/init/init.c
@@ -826,6 +826,14 @@ static void import_kernel_nv(char *name, int for_emulator)
         if (cnt < PROP_NAME_MAX)
             property_set(prop, value);
     }
+      else if (!strncmp(name, "syspart", 7) && name_len > 7) {
+        char prop[PROP_NAME_MAX];
+        int cnt;
+
+        cnt = snprintf(prop, sizeof(prop), "ro.boot.%s", name);
+        if (cnt < PROP_NAME_MAX)
+            property_set(prop, value);
+    }
 }
 
 static void export_kernel_boot_props(void)
